### DISCUSSION TITLE
Server Error Error: Objects are not valid as a React child (found: [object Promise]). If you meant to render a collection of children, use an array instead.

### DISCUSSION THREAD

**Discussion by liaodalin19903** [vercel/next.js#52336] [Help] [answered]
### Summary

I follow the [official document](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching):

```
async function getData() {
  const res = await fetch('https://api.example.com/...')
  // The return value is *not* serialized
  // You can return Date, Map, Set, etc.
 
  // Recommendation: handle errors
  if (!res.ok) {
    // This will activate the closest `error.js` Error Boundary
    throw new Error('Failed to fetch data')
  }
 
  return res.json()
}
 
export default async function Page() {
  const data = await getData()
 
  return <main></main>
}
```

but in my project I get error:

![snapshot](https://github.com/vercel/next.js/assets/12727582/7bbdb24e-cb9e-4c62-8c9d-25f97da366c4)

```
Server Error
Error: Objects are not valid as a React child (found: [object Promise]). If you meant to render a collection of children, use an array instead.

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Call Stack
renderNodeDestructiveImpl
file:///Users/markleo/Desktop/projects/%E4%B8%AA%E4%BA%BA/electron-nextjs-mark-logicmanager/node_modules/react-dom/cjs/react-dom-server.browser.development.js (6179:11)
renderNodeDestructive
file:///Users/markleo/Desktop/projects/%E4%B8%AA%E4%BA%BA/electron-nextjs-mark-logicmanager/node_modules/react-dom/cjs/react-dom-server.browser.development.js (6076:14)
renderIndeterminateComponent
file:///Users/markleo/Desktop/projects/%E4%B8%AA%E4%BA%BA/electron-nextjs-mark-logicmanager/node_modules/react-dom/cjs/react-dom-server.browser.development.js (5785:7)
```

If I remove the `async` of `export default async function Page()` 
and `  const data = await getData()`
it will run again, please tell me how to solve this issue.

**Comment by icyJoseph** [comment]
Are you using the App Router? It looks like you are using an electron-nextjs app, have you migrated to App Router? Or is it using Pages Router?

**Reply by liaodalin19903** [reply]
I don't know how to distinguish app router app / pages router, please tell me how.

**Reply by icyJoseph** [reply]
The App Router is what you use when your application routes are under the `/app` folder, or `src/app`.

Pages route, is what you use when you application routes are under `/pages`, or `src/pages`.

**Reply by liaodalin19903** [reply]
It's under 'renderer/pages', nextjs with electron template.

**Reply by icyJoseph** [reply, marked as answer]
Right, so you have used https://github.com/vercel/next.js/tree/canary/examples/with-electron/renderer, and yeah, under pages the Pages Router kicks in.

![Screenshot 2023-07-07 at 14 38 45](https://github.com/vercel/next.js/assets/21013447/e4952162-b2a6-42d0-8f08-18c807a1bbbc)

The documentation you have shared is for App Router (see the Blue-ish Switch Icon on the Top Left), and how it says `async and await in Server Components`, which are only supported in the App Router.

In Pages Router you can't do that. You need to switch to the pages documentation: https://nextjs.org/docs/pages/building-your-application/data-fetching
